com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -810,60 +810,60 @@ define([
                     });
             }
 
+
             /**
-             * Get job roles based on the passed Contact ID (refresh part of the page)
-             * @param contact_id
-             * @returns {promise}
+             * Fetches the contract ids of the given contact
+             *
+             * @param {int} contactId
+             * @return {Promise} resolves with an array of contract ids
              */
-            function getJobRolesList(contact_id) {
-                var contractsPromise;
-                if (!job_roles.job_contract_ids) {
-                    contractsPromise = HRJobRolesService.getContracts(contact_id).then(function (data) {
-
-                        var job_contract_ids = [];
-                        var contractsData = {};
-
-                        // If we have job contracts, try to get the job roles for the contract
-                        if (data.count != 0) {
-                            for (var i = 0; i < data.count; i++) {
-
-                                // Job contract IDs which will be passed to the "getAllJobRoles" service
-                                job_contract_ids.push(data.values[i]['id']);
-
-                                var contract = {
-                                    id: data.values[i]['id'],
-                                    title: data.values[i]['title'],
-                                    start_date: data.values[i]['period_start_date'],
-                                    end_date: data.values[i]['period_end_date'],
-                                    status: status,
-                                    revisions: data.values[i]['revisions']
-                                };
-
-                                var optionalEndDate = $filter('formatDate')(contract.end_date) || 'Unspecified';
-                                contract.label = contract.title + ' (' + $filter('formatDate')(contract.start_date) + ' - ' + optionalEndDate + ')';
-
-                                contractsData[data.values[i]['id']] = contract;
-                            }
-
-                            // Store the ContractsData what we can reuse later
-                            job_roles.contractsData = contractsData;
+            function contractIdsFromContact(contactId) {
+                return HRJobRolesService.getContracts(contactId).then(function (data) {
+                    var jobContractIds = [];
+                    var contractsData = {};
+
+                    // If we have job contracts, try to get the job roles for the contract
+                    if (data.count !== 0) {
+                        for (var i = 0; i < data.count; i++) {
+                            // Job contract IDs which will be passed to the "getAllJobRoles" service
```

Comment:
(Given this controller is a behemoth of almost 1000 lines and it is in dire need of some refactoring) It doesn't seem so, this method is called only by the original `getJobRolesList`, and it's the only place where `HRJobRolesService.getContracts` is called so there doesn't seem to be a duplication happening here